### PR TITLE
Docs: modify the configuration.md

### PR DESCRIPTION
**Pull request by Zhangg7723**
Modify configuration.md doc with read and write properties

### CODE REVIEW

File: docs/versioned/tables/configuration.md [hunk 1]
```diff
@@ -75,7 +96,6 @@ Iceberg tables support table properties to configure table behavior, like the de
 | commit.manifest-merge.enabled      | true             | Controls whether to automatically merge manifests on writes   |
 | history.expire.max-snapshot-age-ms | 432000000 (5 days) | Default max age of snapshots to keep while expiring snapshots    |
 | history.expire.min-snapshots-to-keep | 1                | Default min number of snapshots to keep while expiring snapshots |
-| history.expire.max-ref-age-ms      | `Long.MAX_VALUE` (forever) | For snapshot references except the `main` branch, default max age of snapshot references to keep while expiring snapshots. The `main` branch never expires. |
```
Comment (jackye1995) [inline docs/versioned/tables/configuration.md:78]: should not be removed

Comment (Zhangg7723) [reply docs/versioned/tables/configuration.md:78]: but I didn't find any usage of this parameter，is it deprecated?

File: docs/versioned/tables/configuration.md [hunk 2]
```diff
@@ -34,29 +34,50 @@ Iceberg tables support table properties to configure table behavior, like the de
...
+| write.manifest-lists.enabled       | true               | Enables to list manifest                                                  |
+| write.delete.isolation-level       | serializable       | Defines the isolation level of write delete                         |
+| write.delete.mode                  | copy-on-write      | Defines the write delete mode                         |
+| write.update.isolation-level       | serializable       | Defines the isolation level of write update             |
+| write.update.mode                  | copy-on-write      | Defines the write update mode                         |
+| write.merge.isolation-level       | serializable       | Defines the isolation level of write merge             |
+| write.merge.mode                  | copy-on-write      | Defines the write merge mode                         |
+| write.upsert.enabled              | false              | Enables the upsert writes                            |
+
+
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:80]: Extra newlines

File: docs/versioned/tables/configuration.md [hunk 3]
```diff
@@ -34,29 +34,50 @@ Iceberg tables support table properties to configure table behavior, like the de
...
+| read.parquet.vectorization.batch-size| 5000            | The batch size of parquet vectorization read           |
+| read.orc.vectorization.enabled    | false              | Enables orc vectorization read                         |
+| read.orc.vectorization.batch-size | 5000               | The batch size of orc vectorization read               |
 
 ### Write properties
 
 | Property                           | Default            | Description                                        |
 | ---------------------------------- | ------------------ | -------------------------------------------------- |
 | write.format.default               | parquet            | Default file format for the table; parquet, avro, or orc |
+| write.delete.format.default               |             | Default delete file format for the table; parquet, avro, or orc |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:47]: Looks like formatting is misaligned in a few places.

File: docs/versioned/tables/configuration.md (hunk 1 above)
Comment (rdblue) [reply docs/versioned/tables/configuration.md:78]: No, it isn't deprecated. We're in the process of adding it.

File: docs/versioned/tables/configuration.md [hunk 4]
```diff
@@ -34,29 +34,50 @@ Iceberg tables support table properties to configure table behavior, like the de
...
 | write.wap.enabled                  | false              | Enables write-audit-publish writes |
 | write.summary.partition-limit      | 0                  | Includes partition-level summary stats in snapshot summaries if the changed partition count is less than this limit |
 | write.metadata.delete-after-commit.enabled | false      | Controls whether to delete the oldest version metadata files after commit |
 | write.metadata.previous-versions-max       | 100        | The max number of previous version metadata files to keep before deleting after commit |
 | write.spark.fanout.enabled       | false        | Enables Partitioned-Fanout-Writer writes in Spark |
+| write.object-storage.enabled      | false              | Enables object storage write like s3              |
+| write.data.path                    | a "data" folder underneath the root path of the table | Defines the path of data files |
+| write.metadata.path                | a "metadata" folder underneath the root path of the table | Defines the path of metadata files |
+| write.manifest-lists.enabled       | true               | Enables to list manifest                                                  |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:71]: This should be removed. This is a compatibility property and not something people should use.

File: docs/versioned/tables/configuration.md [hunk 5]
```diff
@@ -34,29 +34,50 @@ Iceberg tables support table properties to configure table behavior, like the de
...
 | write.distribution-mode            | none               | Defines distribution of write data: __none__: don't shuffle rows; __hash__: hash distribute by partition key ; __range__: range distribute by partition key or sort key if table has an SortOrder |
+| write.delete.distribution-mode     | hash               | Defines distribution of write delete data           |
 | write.wap.enabled                  | false              | Enables write-audit-publish writes |
 | write.summary.partition-limit      | 0                  | Includes partition-level summary stats in snapshot summaries if the changed partition count is less than this limit |
 | write.metadata.delete-after-commit.enabled | false      | Controls whether to delete the oldest version metadata files after commit |
 | write.metadata.previous-versions-max       | 100        | The max number of previous version metadata files to keep before deleting after commit |
 | write.spark.fanout.enabled       | false        | Enables Partitioned-Fanout-Writer writes in Spark |
+| write.object-storage.enabled      | false              | Enables object storage write like s3              |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:68]: This is incorrect. S3 is enabled by using `s3://` URIs.

File: docs/versioned/tables/configuration.md [hunk 6]
```diff
@@ -34,29 +34,50 @@ Iceberg tables support table properties to configure table behavior, like the de
...
+| write.data.path                    | a "data" folder underneath the root path of the table | Defines the path of data files |
+| write.metadata.path                | a "metadata" folder underneath the root path of the table | Defines the path of metadata files |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:70]: These are still subject to `LocationProvider` decisions.

I wouldn't say "defines". This is where the default location provider will place metadata files.

File: docs/versioned/tables/configuration.md [hunk 7]
```diff
@@ -34,29 +34,50 @@ Iceberg tables support table properties to configure table behavior, like the de
...
+| write.manifest-lists.enabled       | true               | Enables to list manifest                                                  |
+| write.delete.isolation-level       | serializable       | Defines the isolation level of write delete                         |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:72]: The isolation level properties should define the different levels if you want to include them here.

File: docs/versioned/tables/configuration.md [hunk 8]
```diff
@@ -34,29 +34,50 @@ Iceberg tables support table properties to configure table behavior, like the de
...
+| write.delete.mode                  | copy-on-write      | Defines the write delete mode                         |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:73]: The mode properties should state what the two modes are (copy-on-write and merge-on-read) and should state when merge-on-read can be used (v2 tables only)

File: docs/versioned/tables/configuration.md [hunk 9]
```diff
@@ -34,29 +34,50 @@ Iceberg tables support table properties to configure table behavior, like the de
...
+| write.update.isolation-level       | serializable       | Defines the isolation level of write update             |
+| write.update.mode                  | copy-on-write      | Defines the write update mode                         |
+| write.merge.isolation-level       | serializable       | Defines the isolation level of write merge             |
+| write.merge.mode                  | copy-on-write      | Defines the write merge mode                         |
+| write.upsert.enabled              | false              | Enables the upsert writes                            |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:78]: This is a Flink property, so it shouldn't go in this table. Upserts are not generally supported.

File: docs/versioned/tables/configuration.md [hunk 10]
```diff
@@ -102,6 +122,9 @@ Iceberg catalogs support using catalog properties to configure catalog behaviors
 | warehouse                         | null               | the root path of the data warehouse                    |
 | uri                               | null               | a URI string, such as Hive metastore URI               |
 | clients                           | 2                  | client pool size                                       |
+| cache-enabled                     | true               | Controls whether the catalog will cache table entries upon load. |
+| cache.expiration-interval-ms      | 30000              | Controls the duration for which entries in the catalog are cached. |
+| client.pool.cache.eviction-interval-ms | 5000          | Controls the duration for which clients in the pool          |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:127]: I don't understand this description.

File: docs/versioned/tables/configuration.md [hunk 11]
```diff
@@ -102,6 +122,9 @@ Iceberg catalogs support using catalog properties to configure catalog behaviors
 | warehouse                         | null               | the root path of the data warehouse                    |
 | uri                               | null               | a URI string, such as Hive metastore URI               |
 | clients                           | 2                  | client pool size                                       |
+| cache-enabled                     | true               | Controls whether the catalog will cache table entries upon load. |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:125]: No need for punctuation at the end of descriptions

File: docs/versioned/tables/configuration.md [hunk 12]
```diff
@@ -102,6 +122,9 @@ Iceberg catalogs support using catalog properties to configure catalog behaviors
...
+| cache.expiration-interval-ms      | 30000              | Controls the duration for which entries in the catalog are cached. |
```
Comment (rdblue) [inline docs/versioned/tables/configuration.md:126]: I think "entries" should be "tables".

File: docs/versioned/tables/configuration.md (hunk 2 above)
Comment (Zhangg7723) [reply docs/versioned/tables/configuration.md:80]: @rdblue Thanks for all your earnest reviews，I will double check my commit。

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @Zhangg7723!

**Pull request merged**